Makefile

```diff
@@ -114,521 +105,312 @@ endef
+        printf "%s\t[ ", $$1; \
+        if ($$2 == "OK") \
+            color = 28;   \
+        else \
+            color = 196;   \
+        system("tput setaf " color "; printf " $$2 "; tput sgr0"); \
+        printf " ]\n"; \
+    }' | column --separator $$'\t' --table
+# Step 3/3 of expected
+expected: check
+	rm -rf expected/$(BUILD_DIR) > /dev/null 2>&1 || true
+	$(muffle)mkdir -p expected
+	cp -r $(BUILD_DIR) expected/$(BUILD_DIR)
+
+force-extract:
+	mkdir -p /tmp/sotn-decomp
+	rm -rf /tmp/sotn-decomp/src || true; mv src /tmp/sotn-decomp/src
+	find $(BUILD_DIR) -type f -name "*.ld" -delete || true
+	$(MAKE) extract
+	rm -rf $(SRC_DIR)
+	mv /tmp/sotn-decomp/src $(SRC_DIR)
+force-extract-disk:
+	@rm -rf $(EXTRACTED_DISK_DIR) || true
+	$(MAKE) extract-disk
+# This is currently intentionally hard coded to us because the us symbols files are used for finding functions in other versions
+$(addsuffix .force-symbols,$(notdir $(wildcard $(BUILD_DIR:$(VERSION)=us)/*.elf))): %.elf.force-symbols: | $(VENV_DIR)
+	$(PYTHON) tools/symbols.py elf $(BUILD_DIR)/$*.elf > $(CONFIG_DIR)/symbols$(if $(filter-out stmad,$*),.us).$*.txt
+force-symbols: $(addsuffix .force-symbols,$(notdir $(wildcard $(BUILD_DIR:$(VERSION)=us)/*.elf)))
+
+context: $(M2CTX_APP) | $(VENV_DIR)
+ifndef SOURCE
+	$(error SOURCE environment variable must be set to generate context)
+endif
+	VERSION=$(VERSION) $(PYTHON) $(M2CTX_APP) $(SOURCE)
+	$(call echo,ctx.c has been updated.)
+
+disks/us: | $(SOTNDISK)
+	$(SOTNDISK) extract disks/sotn.$(VERSION).cue $(EXTRACTED_DISK_DIR)
+disks/pspeu:
+	mkdir -p $(EXTRACTED_DISK_DIR)
+	7z x -y disks/sotn.pspeu.iso -o$(EXTRACTED_DISK_DIR)
+disks/saturn:
+	bchunk disks/sotn.$(VERSION).bin disks/sotn.$(VERSION).cue disks/sotn.$(VERSION).iso
+	7z x disks/sotn.$(VERSION).iso01.iso -o$(EXTRACTED_DISK_DIR) || true
+extract-disk: $(EXTRACTED_DISK_DIR)
+
+ovl_to_bin  = $(call to_upper,$(call add_ovl_prefix,$(1),ST/,BOSS/)/$(1).BIN $(call add_ovl_prefix,$(1),ST/,BOSS/)/F_$(1).BIN)
+DISK_PREPARE_FILES := DRA.BIN BIN/RIC.BIN ST/SEL/SEL.BIN $(addprefix SERVANT/,$(call to_upper,$(addsuffix .BIN,$(SERVANTS))))
+DISK_PREPARE_FILES += $(foreach target,$(filter-out sel,$(STAGES)) $(BOSSES),$(call ovl_to_bin,$(target)))
+disk-prepare: build $(SOTNDISK)
+	mkdir -p $(BUILD_DISK_DIR); cp -r $(EXTRACTED_DISK_DIR)/* $(BUILD_DISK_DIR)
+	cp $(BUILD_DIR)/main.exe $(BUILD_DISK_DIR)/SLUS_000.67
+	$(call echo,cp $(BUILD_DIR)/*.BIN $(BUILD_DISK_DIR)/) $(foreach item,$(DISK_PREPARE_FILES),cp $(BUILD_DIR)/$(notdir $(item)) $(BUILD_DISK_DIR)/$(item);)
+disk-debug: disk-prepare
+	cd tools/sotn-debugmodule && $(MAKE)
+	cp $(BUILD_DIR:$(VERSION)=)/sotn-debugmodule.bin $(BUILD_DISK_DIR)/SERVANT/TT_000.BIN
+	$(SOTNDISK) make $(BUILD_DIR:$(VERSION)=)/sotn.$(VERSION).cue $(BUILD_DISK_DIR) $(CONFIG_DIR)/disk.$(VERSION).lba
+disk: disk-prepare
+	$(SOTNDISK) make $(BUILD_DIR:/$(VERSION)=)/sotn.$(VERSION).cue $(BUILD_DISK_DIR) $(CONFIG_DIR)/disk.$(VERSION).lba
```

Comment:
I redid the disk targets because it was already unwieldy and it will only get worse as more overlays are added.  This has main, dra, ric, and sel hardcoded because there was no benefit to dynamically building those.  It uses the enabled build targets to add all of the other overlays.

I did just realize that the two weapon .BIN files are not accounted for in here (they aren't in master either).